llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -10975,75 +10975,116 @@ SDValue RISCVTargetLowering::lowerVECTOR_DEINTERLEAVE(SDValue Op,
   assert(VecVT.isScalableVector() &&
          "vector_interleave on non-scalable vector!");
 
+  const unsigned Factor = Op->getNumValues();
+
   // 1 bit element vectors need to be widened to e8
   if (VecVT.getVectorElementType() == MVT::i1)
     return widenVectorOpsToi8(Op, DL, DAG);
 
-  // If the VT is LMUL=8, we need to split and reassemble.
-  if (VecVT.getSizeInBits().getKnownMinValue() ==
+  // If concatenating would exceed LMUL=8, we need to split.
```

Comment:
This code doesn't handle fixed vectors which weren't possible with Factor==2, because SelectionDAGBuilder turns them into shuffles. But the SelectionDAGBuilder in this patch doesn't use shuffle for fixed vectors for factor 3, 5 or 7.

Reply:
Done